### ISSUE TITLE
Top level multi-line tuples/lists are formatted to unparseable code 

### ISSUE THREAD

**Issue by amesgen** [tweag/ormolu#778]
**Describe the bug**
The snippet
```haskell
(a
 ,b)

[a
 ,b]
```
is formatted (with `--unsafe`) to
```haskell
( a,
  b
)

[ a,
  b
]
```
which can not be parsed. The closing parens/brackets can't be in the first column.

**Environment**
Ormolu 0.2.0.0

**Comment by mrkkrp** [comment]
But what a top-level tuple or a list mean?

**Comment by amesgen** [comment]
1. People could write very weird `TH.Lift` instances. :grin:
 2. A bit less contrived: Consider
    ```haskell
    test =
      (1
       ,2)
    ```
    which one might try to format via `ormolu --start-line 2` (which will fail right now, even though the file can be formatted as a whole).

**Comment by mrkkrp** [comment]
Right, so it's mostly important in the context of region formatting.

**Comment by mrkkrp** [comment]
The solution here would be to identify these situations and shift the closing parenthesis or brace like we already do in certain situations.

**Comment by buggymcbugfix** [comment]
> Right, so it's mostly important in the context of region formatting.

Not exactly.

> But what a top-level tuple or a list mean?

This is just a pattern. Consider:

```hs
a, b :: Int
(a, b) = (1, 2)
```

Another example of a multiline pattern might be

```hs
a, b :: Int
Foo
  a
  b = mkFoo 42
```

Apparently for multiline patterns the subsequent lines must always be indented more than the first column of the first line of the pattern.

**Comment by amesgen** [comment]
@buggymcbugfix The cases you mention are already correctly handled by Ormolu, or I am missing something in your comment?

**Comment by buggymcbugfix** [comment]
I didn't check if they are correctly handled, but I assume they would be. I was merely answering @mrkkrp's question.

**Comment by amesgen** [comment]
Yeah, but the cases you mention are not top-level tuples/lists, but different constructs. But the asymmetry between tuple expressions and tuple patterns is indeed surprising:

```haskell
a = (a
    ,a)
```
is valid (even with less indentation in the second line), but
```haskell
(a
,a) = a
```
is not.

**Comment by gkaracha** [comment]
I thought I'd look into this issue but I've been unable to come up with a snippet that compiles but ormolu formats into code that no longer does. Is there a minimal example (that compiles) that I could use to trigger the offending behavior?
